### PR TITLE
[CPROD-1016] Move out auction api out of `is20` to `canister-sdk`

### PR DESCRIPTION
**Pull request by awkure**

### CODE REVIEW

File: dfx.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "token_factory": {
       "build": "bash scripts/build.sh",
       "candid": "src/candid/token-factory.did",
-      "wasm": "target/wasm32-unknown-unknown/release/token-factory.wasm",
+      "wasm": "target/wasm32-unknown-unknown/release/factory.wasm",
```
Comment (blutooth) [inline dfx.json:6]: I wouldnt change this, likely to break a lot of scripts.

File: src/token/api/src/canister.rs [hunk 2]
```diff
@@ -146,7 +138,7 @@ pub trait TokenCanisterAPI: Canister + Sized {
         TokenInfo {
             metadata: self.state().borrow().get_metadata(),
             feeTo: fee_to,
```
Comment (rumatoest) [inline src/token/api/src/canister.rs:140]: Please fix here other camelCase properties to the snake case in this structure

File: src/token/api/src/canister/erc20_transactions.rs [hunk 3]
```diff
@@ -310,7 +313,7 @@ mod tests {
                 owner: alice(),
                 fee: Tokens128::from(0),
                 feeTo: alice(),
```
Comment (rumatoest) [inline src/token/api/src/canister/erc20_transactions.rs:315]: snake case please

**Review by rumatoest — CHANGES_REQUESTED** [review]
It is better to rename all fields in structs to snake case, not just some of them.

File: spec/IS20.md [hunk 4]
```diff
@@ -70,7 +70,7 @@ type Metadata = {
    name : text; // token name
    symbol : text; // token symbol
    decimals : nat8; // token decimal
-   totalSupply : Tokens128; // token total supply
+   total_supply : Tokens128; // token total supply
```
Comment (itsyaasir) [inline spec/IS20.md:73]: remove this, we don't have `total_supply` in `Metadata`

**Comment by awkure** [comment]
Oh, this PR is still open

File: src/token/impl/src/canister.rs [hunk 5]
```diff
@@ -1,45 +1,90 @@
...
+                seconds: DEFAULT_AUCTION_PERIOD_SECONDS,
+            },
+            ic_canister::ic_kit::ic::caller(),
+        ));
+    }
+
+    #[pre_upgrade]
+    fn pre_upgrade(&self) {
+        let token_state = Rc::<RefCell<CanisterState>>::try_unwrap(self.state())
+            .expect("Someone has the token factory state borrowed.")
```
Comment (hagsteel) [inline src/token/impl/src/canister.rs:62]:
```suggestion
            .expect("token factory state already borrowed.")
```

Comment (hagsteel) [reply src/token/impl/src/canister.rs:62]: This applies for all the other ones with the same message.

**Review by hagsteel — CHANGES_REQUESTED** [review]

**Review by itsyaasir — APPROVED** [review]

**Pull request closed without merging**